### PR TITLE
systemd-qa-tests: add binary journal log upload

### PR DESCRIPTION
**Pull request by tblume**
The binary format is much more powerful for a deeper analysis.
One can easily inspect all logs emitted by the containers running on the
system, filter messages, change the ouput format, etc...
Still the text representation should stay as well to be able to take a
direct look from within the browser.

- Verification run: http://knopfler.arch.suse.de/tests/56#downloads

### CODE REVIEW

File: lib/systemd_testsuite_test.pm [hunk 1]
```diff
@@ -123,7 +123,13 @@ sub post_fail_hook {
     my ($self) = @_;
     #upload logs from given testname
     $self->tar_and_upload_log('/var/opt/systemd-tests/logs', '/tmp/systemd_testsuite-logs.tar.bz2');
-    $self->save_and_upload_log('journalctl --no-pager -axb -o short-precise', 'journal.log');
+    if ((script_run 'test -d /var/log/journal')) {
+        $self->tar_and_upload_log('/var/log/journal', 'binary-journal-log.tar.bz2');
```
Comment (foursixnine) [inline lib/systemd_testsuite_test.pm:127]: Why not both in the same tar call? 
```suggestion
        $self->tar_and_upload_log('/var/log/journal /run/log/journal', 'binary-journal-log.tar.bz2');
```

Comment (tblume) [reply lib/systemd_testsuite_test.pm]: Indeed, that looks more elegant and it seems the code doesn't error out when one of the directories isn't present.
I've updated the pr, thanks.

**Comment by foursixnine** [comment]
@tblume could you rebase? I totally missed this pr!

**Comment by tblume** [comment]
> @tblume could you rebase? I totally missed this pr!

Ok, done.

**Comment by foursixnine** [comment]
@tblume tidy is not your friend xD: https://travis-ci.org/github/os-autoinst/os-autoinst-distri-opensuse/jobs/717565583#L481

`make tidy-full` should do the trick :)

**Comment by tblume** [comment]
> @tblume tidy is not your friend xD: https://travis-ci.org/github/os-autoinst/os-autoinst-distri-opensuse/jobs/717565583#L481
> 
> `make tidy-full` should do the trick :)

Now tidy seems to like me more. :)

**Pull request merged**